### PR TITLE
Allow gridsearch to be called from a project which imports `kymata` as a dependency

### PR DESCRIPTION
**Pull request by caiw**
## Necessary changes for packaging fixes

- Moving the `invokers` scripts inside the `kymata` package.
- Moving the core data needed by the gridsearch inside the `kymata` package.

## Other updates in this PR

- Gridsearch can load config files from outside of `kymata-core` (does not change previous behaviour).
- Gridsearch can load function contours from outside prescribed data dirs (does not change previous behaviour).
- Functions and data can be separately resampled.
  - Function sample rate is specified at the CLI using `--function-sample-rate <sample_rate_hz>`. Default is `1000` (Hz) as previously.
  - Separately, data (and function) can be downsampled using `--resample <sample_rate_hz>`.
  - This replaces the previous `--downsample <integer_ratio>` argument, but the behaviour on the data is the same. I.e. `--downsample 5` (subsample ratio) is now `--resample 200` (Hz), and applies to both function and data.
    - Function is directly resampled to target rate, not resampled "current → 1000 → target" as before, which is slightly more accurate.
  - Dataset sample rate specified in config files instead of invisible hard-coded value.
  - Extra checks added to make sure that function and data have the same sample rate, with logging and errors where appropriate.
  - New tests for function resampling code.

## Formatting changes

- A lot of unnecessary whitespace in CLI arg definitions removed from previous autoformatting. Sorry this makes for a big diff, but it's (imo) much easier to read.

### CODE REVIEW

**Comment by neukym** [comment]
> A lot of unnecessary whitespace in CLI arg definitions removed from previous autoformatting. Sorry this makes for a big diff, but it's (imo) much easier to read.

100% agree with this - I might go through removing whitespace everywhere at some point.

**Comment by neukym** [comment]
A random issue with this branch - can you check the output location? It seems to be outputting to `kymata-core/output/` not `kymata-core/kymata-core-data/output/` as it used to previously? But I'm not sure how this has happened, I don't think you have changed this in the code.

**Comment by caiw** [comment]
> A random issue with this branch - can you check the output location? It seems to be outputting to `kymata-core/output/` not `kymata-core/kymata-core-data/output/` as it used to previously? But I'm not sure how this has happened, I don't think you have changed this in the code.

In the code this is picked using `Path(data_root_path(), "output")`, and afaics the default location returned by `data_root_path()` (if not overridden by the user) is by default `$KYMATA_DATA_ROOT` and otherwise is `kymata-core/kymata-core-data/`. Is it possible you have `$KYMATA_DATA_ROOT` set in the shell you're running this branch in?

For example if you're using [`kymata-analyses/analysi-visual-quadrant`](https://github.com/kymata-atlas/kymata-analyses/tree/main/analysis-visual-quadrant), `$KYMATA_DATA_ROOT` is [set in the gridsearch script](https://github.com/kymata-atlas/kymata-analyses/blob/bc163fddaab0d26a81017378ce9db0107dea5afa/analysis-visual-quadrant/analysis/gridsearch_colvect_models.sh#L20).

**Comment by caiw** [comment]
(Improved `data_root_path()` docstring)

**Comment by neukym** [comment]
> In the code this is picked using `Path(data_root_path(), "output")`, and afaics the default location returned by `data_root_path()` (if not overridden by the user) is by default `kymata-core/kymata-core-data/`. One way to override it is by setting `$KYMATA_DATA_ROOT` — is it possible you have that set in the shell you're running this branch in?
> 
> For example if you're using [`kymata-analyses/analysi-visual-quadrant`](https://github.com/kymata-atlas/kymata-analyses/tree/main/analysis-visual-quadrant), `$KYMATA_DATA_ROOT` is set [in the gridsearch script](https://github.com/kymata-atlas/kymata-analyses/blob/bc163fddaab0d26a81017378ce9db0107dea5afa/analysis-visual-quadrant/analysis/gridsearch_colvect_models.sh#L20).

Ah, yes, that would have been it. Thanks for checking.

**Review by neukym — APPROVED** [review]
Looks good, happy to merge.

File: kymata/invokers/run_gridsearch.py [hunk 1]
```diff
@@ -36,149 +34,80 @@ def main():
...
+    parser.add_argument("--use-inverse-operator", action="store_true",
+                        help="Use inverse operator to conduct gridsearch in source space.")
+    parser.add_argument("--morph", action="store_true",
+                        help="Morph hexel data to fs-average space prior to running gridsearch. "
+                             "Only has an effect if an inverse operator is specified.",)
+    parser.add_argument("--inverse-operator-suffix", type=str, default="_ico5-3L-loose02-cps-nodepth-fusion-inv.fif",
+                        help="inverse solution suffix")
+
+    parser.add_argument("--snr", type=float, default=3, help="Inverse solution SNR")
+    parser.add_argument("--resample", type=float, required=False, default=None, help="Resample rate in Hz.")
```
[the remark this answers is not in the document]
Comment (neukym) [reply kymata/invokers/run_gridsearch.py:77]: actually, can we do this first?

[the remark this answers is not in the document]
Comment (neukym) [reply kymata/invokers/run_gridsearch.py]: done.

**Pull request merged**